pulumi/python/infrastructure/digitalocean/domk8s/__main__.py

```diff
@@ -15,7 +15,7 @@
     node_count = 3
 k8s_version = config.get('k8s_version')
 if not k8s_version:
-    k8s_version = '1.21.11-do.0'
+    k8s_version = '1.22.8-do.0'
```

Comment:
**Question**: Is there a need to make sure that versions of the DO K8S version stays in sync with the `.sample` file above?

Reply:
I try to keep them in sync, but if you take a look at #97 one of my plans is to have the bulk of the config to be processed by the startup script so we can stop specifying the values as fallbacks in the code *and* maintaining the sample script. Any thoughts you have on how best to accomplish this are welcome; I've not progressed far beyond the "this kinda is not a great pattern and is a pain" stage.

Reply:
I think there's two points where clarity and consistency are important (and I may be just restating what you said above but this is my understanding):

1. On install.  While installing I want to be clear on the versions of the primary components over which I have control.
2. When making changes or troubleshooting.  I want to know where I can go and find the one source of truth for versions of each component I control (and maybe read only the versions of some I don't)

As for how we accomplish that, I'd need to get more familiar with MARA to have a good opinion, but having a general config file for MARA itself seems like a reasonable idea that's common to many such frameworks.  Like having a `.maraconf` file that you plop in the main directory that could be generated off of your sample.

This actually reminds me of an old [blog post](https://medium.com/@elijahz/what-version-is-your-infrastructure-3a61fe804d0e) @dekobon wrote.  But I'm not sure if his thinking has changed since then 😅